src/components/core/badge/badge.tsx

```diff
@@ -0,0 +1,118 @@
+import classNames from 'classnames';
+import { forwardRef, HTMLAttributes } from 'react';
+import { styled } from 'Styles/stitches.config';
+
+type TBadgeSize = 'small' | 'medium' | 'large';
+type TBadgeSpacing = 'tight' | 'loose';
+type TVisiblity = 'icon-only' | 'label-only' | 'icon-and-label';
+type TLabel = 'bold' | 'regular';
+
+export interface BadgeProps extends HTMLAttributes<HTMLSpanElement> {
+    icon_src?: string;
+    icon_class?: string;
+    size?: TBadgeSize;
+    spacing?: TBadgeSpacing;
+    visiblity?: TVisiblity;
+    label?: TLabel;
+    iconAlt?: string;
+}
+
+const BadgeContainer = styled('span', {
+    backgroundColor: '$greyLight200',
+    borderRadius: '$default',
+    fontWeight: '$regular',
+    display: 'inline-flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    variants: {
+        size: {
+            small: {
+                minHeight: '24px',
+                fontSize: '$4xs',
+                '@mobile': {
+                    '& .badge--icon': {
+                        height: '8px',
+                    },
+                    minHeight: '20px',
+                    fontSize: '$5xs',
+                },
+            },
+            medium: {
+                minHeight: '26px',
+                fontSize: '$3xs',
+                '@mobile': {
+                    minHeight: '22px',
+                    fontSize: '$4xs',
+                },
+            },
+            large: {
+                minHeight: '28px',
+                fontSize: '$2xs',
+                '@mobile': {
+                    minHeight: '24px',
+                    fontSize: '$3xs',
+                },
+            },
+        },
+        padding: {
+            tight: {
+                padding: '0px 4px',
+                '& .badge--icon__padding': {
```

Comment:
```suggestion
                '& .badge--icon__padding': {
```
Related to comment [1]

Reply:
removed these and used compoundVariables. thanks :pray: